jwst/coron/smooth_psf.py

```diff
@@ -0,0 +1,78 @@
+"""Replace bad pixels with the median of the surrounding pixel and median fill
+   smooth the psf images.
+ """
+import logging
+import numpy as np
+
+from scipy.ndimage import median_filter
+
+from jwst.datamodels import dqflags
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+def median_fill_value(input_array, input_dq_array, bsize, xc, yc):
+    """
+        Arguments:
+        ----------
+        input_array : ndarray
+            Input array to filter.
+        input_dq_array : ndarray
+            Input data quality array
+        bsize : scalar
+            box size of the data to extract
+        xc: scalar
+           x position of the data extraction
+        xc: scalar
+           y position of the data extraction
+        """
+    # set the half box size
+    hbox = int(bsize/2)
+
+    # Extract the region of interest for the data
+    try:
+        data_array = input_array[xc - hbox:xc + hbox, yc - hbox: yc + hbox]
+        dq_array = input_dq_array[xc - hbox:xc + hbox, yc - hbox: yc + hbox]
+    except IndexError:
+        # If the box is outside the data return 0
+        log.warning('Box for median filter is outside the data.')
+        return 0.
+
+    filtered_array = data_array[dq_array != dqflags.pixel['DO_NOT_USE']]
+    median_value = np.median(filtered_array)
```

Comment:
Is it expected that `filtered_array` doesn't have `NaN` values at all? Otherwise, could use `np.nanmedian`

Reply:
The DO_NOT_USE flag should cover the case for NaN's

Reply:
Ah, ok, then in that case, when would `np.median` return `NaN`?

Reply:
As long as the flag is accurate np.median will return a Nan if the no values are left after the flagging. np.nanmedian will also return a NaN in that case. 
Do you have another case in mind?

Reply:
No, I don't think so! I was trying to figure out whether or not we could be more explicit in trying to catch any issues here